src/mesh/unstructured_quad_mesh.jl

```diff
@@ -0,0 +1,208 @@
+"""
+    UnstructuredQuadMesh{NDIMS, RealT<:Real} <: AbstractMesh{NDIMS}
+
+An unstructured (possibly curved) quadrilateral mesh.
+
+All mesh information, neighbour coupling, and boundary curve information is read in from a mesh file
+
+!!! warning "Experimental code"
+    This mesh type is experimental and can change any time.
+"""
+struct UnstructuredQuadMesh{NDIMS, RealT<:Real} <: AbstractMesh{NDIMS}
+  filename             ::String
+  n_corners            ::Int64
+  n_interfaces         ::Int64
+  n_boundary           ::Int64
+  n_elements           ::Int64
+  poly_deg             ::Int64
+  corners              ::Array{RealT, 2}  # [ndims, n_corners]
+  neighbour_information::Array{Int64, 2}  # [neighbour node/element/edge ids, n_interfaces]
+  boundary_names       ::Array{String, 2} # [local sides, n_elements]
+  periodicity          ::Bool
+  element_node_ids     ::Array{Int64, 2} # [node ids, n_elements]
+  element_is_curved    ::Vector{Bool}
+  elements_curves      ::Array{GammaCurve, 2} # [local sides, n_elements]
+end
```

Comment:
This is very specific to the 2D version, especially because of the `GammaCurve`type. I will change this

Reply:
If I change this how would I access the value of `NDIMS` when needed, e.g., in `ndofs`?

Reply:
You could, e.g., implement `@inline ndims(::UnstructuredQuadMesh) = 2`

Reply:
Hmm, changing this causes some issues for me in the `wrap_array` routines in `dg_common.jl`. I think it is because I removed the `NDIMS` type parameter.

Reply:
Right. You could try changing `UnstructuredQuadMesh{2}` to just `UnstructuredQuadMesh` since it is defined to work for 2D only anyways.